File: backend/app/schemas/attachment.py
# SPDX-FileCopyrightText: 2025 Weibo, Inc.
#
# SPDX-License-Identifier: Apache-2.0

"""
Attachment schemas for API responses and request validation.
"""

from typing import Optional

from pydantic import BaseModel


class TruncationInfo(BaseModel):
    """Information about content truncation."""

    is_truncated: bool = False
    original_length: Optional[int] = None
    truncated_length: Optional[int] = None
    truncation_message_key: Optional[str] = None  # i18n key for frontend


class AttachmentResponse(BaseModel):
    """Response model for attachment operations."""

    id: int
    filename: str
    file_size: int
    mime_type: str
    status: str
    text_length: Optional[int] = None
    error_message: Optional[str] = None
    error_code: Optional[str] = None  # Error code for i18n mapping
    truncation_info: Optional[TruncationInfo] = None

    class Config:
        from_attributes = True


class AttachmentDetailResponse(AttachmentResponse):
    """Detailed response model including subtask_id."""

    subtask_id: Optional[int] = None
    file_extension: str
    created_at: str

File: frontend/src/utils/attachment.ts
// SPDX-FileCopyrightText: 2025 Weibo, Inc.
//
// SPDX-License-Identifier: Apache-2.0

export const ATTACHMENT_SCHEME = 'attachment://'

export function parseAttachmentSchemeUrl(url?: string | null): number | null {
  if (!url) {
    return null
  }

  const trimmedUrl = url.trim()
  if (!trimmedUrl.startsWith(ATTACHMENT_SCHEME)) {
    return null
  }

  const match = trimmedUrl.match(/^attachment:\/\/(\d+)/)
  if (!match) {
    return null
  }

  const attachmentId = Number(match[1])
  if (!Number.isSafeInteger(attachmentId) || attachmentId <= 0) {
    return null
  }

  return attachmentId
}
